js/cardview.js

```diff
@@ -16,28 +16,120 @@ let Card = React.createClass({
+  // style the card based on if it has attacked, is castable, etc
+  cssClassesForCard: function(card) {
+    let cssClassCanPlay = '';
     if (this.props.cardInfo.cost > this.props.player.mana) {
+      cssClassCanPlay = "too-expensive";
+    }
+    let cssClass = this.props.cardInfo.hasFocus ? 
+                   'playing-card active-card' : 'playing-card';
+    let cssClassAttacked = this.props.cardInfo.hasAttacked ? 
+                           'has-attacked-card' : '';
+    let cssClassJustPlayed = this.props.cardInfo.enteredPlayThisTurn ? 
+                             'just-played-card' : '';
+    let combinedCSS = cssClass + ' ' + 
+                      cssClassCanPlay + ' ' + 
+                      cssClassAttacked + ' ' + 
+                      cssClassJustPlayed; 
+    return combinedCSS;
+  },
+
+  // click cards in current player's hand or board
+  clickCard: function() {
+    
+    // ATTACK
+    // can click opponent's in play cards
+    // after player has clicked his own in play card to attack
+    var hasActiveCard = false;
+    var fromAttackIndex = 0;
+    var attackingCard;
+    for (var card of window.game.current().board) {
+      if (card.hasFocus) {
+        attackingCard = card;
+        hasActiveCard = true;
+        break;
+      }
+      fromAttackIndex++;
+    }
+    var toIndex = window.game.opponent().board.indexOf(this.props.cardInfo);
+    if (toIndex != -1 && hasActiveCard) {
+      this.attack(fromAttackIndex, toIndex, attackingCard)
+      return;      
+    }
+
+    // PLAY CARD FROM HAND
+    // can click to play cards in hand
+    var fromIndex = window.game.current().hand.indexOf(this.props.cardInfo);
+    if (fromIndex != -1) {
+      this.playFromHand(fromIndex, this.props.cardInfo);
       return;
     }
     
-    this.props.hasFocus = !this.props.hasFocus;
-    // play card on 2nd click
-    if (!this.props.hasFocus) {
-      var fromIndex = window.game.current().hand.indexOf(this.props.cardInfo);
-      console.log(fromIndex);
-      console.log(window.game.current().hand);
-      var move = {"op":"play", 
+    // HIGHLIGHT CARD, OR SMASH FACE IF ALREADY HIIGHLIGHTED
+    // can click a card in play if it's in the player's board
+    // and it's not summoning sick or already attacked
+    fromIndex = window.game.current().board.indexOf(this.props.cardInfo);
+    if (fromIndex != -1 && 
+        !this.props.cardInfo.enteredPlayThisTurn &&
+        !this.props.cardInfo.hasAttacked) {
+      this.clickCardInPlay(fromIndex, this.props.cardInfo);
+      return;
+    }
+
+    // if it's not in their hand or board, then do nothing
+
+  },
+  
+  // smash creature's face
+  attackCreature: function(fromAttackIndex, toIndex, attackingCard) {
+    var move = {"op":"attack", 
+                "from":fromAttackIndex,
+                "to": toIndex
+               };
+    window.client.makeLocalMove(move);
+    window.client.gameView.forceUpdate();
```

Comment:
this will double force update since makeLocalMove handles that. you should never need to call forceUpdate from a view directly like this.